### ISSUE TITLE
Trying to shred multiple files in Windows Explorer crashes BleachBit 4.2.0 (and 4.0.0)

### ISSUE THREAD

**Issue by armbck** [bleachbit/bleachbit#1080]
Hi! Thank you for BleachBit, it is a useful software. I've been waiting for a fix for this problem in version 4.2.0, but as it did not come I'll open this bug.

How to reproduce the crash: 
1. Open Windows Explorer
2.  select two or more files with mouse
3.  open context menu with right mouse button and choose Shred with BleachBit
=> BleachBit starts the process but dies right away. The files are not shred. 

Actually in 4.2.0 BleachBit dies even with one file. In 4.0.0 it managed to shred one file using Win Explorer context menu.

If I first open BleachBit, choose Shred files and choose multiple files in File Open -window, then it works, but it takes so many more steps that way.

**Comment by abitrolly** [comment]
Without access to Windows machine I can not reproduce to see what happens. BleachBit should produce an error trace in this case, but I am not sure if it is written by default and how to turn it on for the Explorer menu.

**Comment by armbck** [comment]
Thank you for your quick reply @abitrolly. You are right, I could not either figure a way to show debug messages for this case. I switched the "Show Debug Messages" option on and tried two ways. 1) BleachBit open -> nothing happened and no debug messages were shown 2) BleachBit closed - BleachBit tries to open, but closes immediately. Tell me if you find a way to debug without a development environment. Thanks.

**Comment by abitrolly** [comment]
The only instruction I could find it here https://docs.bleachbit.org/doc/troubleshooting.html but it is unlikely to help if BleachBit works normally without the Explorer menu. The instruction there tells to run this in command on console manually.
```
"%ProgramFiles(x86)%\BleachBit\bleachbit_console.exe" --gui --debug-log=%USERPROFILE%\desktop\bleachbit.log
```
For the Explorer, however, there must be a way to find and modify the menu item. Maybe @az0 or somebody else know where the menu item is and what its contents should look like.

An alternative could be to set some kind of `DEBUG_LOG` environment variable, but from the code I don't see that BleachBit supports this.

**Comment by az0** [comment]
Selecting multiple files in the context menu for shredding does not work because of a limitation in Windows, though last I checked, there was not a crash. Here is an [old bug ticket from 2013](https://bugs.launchpad.net/bleachbit/+bug/1261252). See [this comment ](https://github.com/bleachbit/bleachbit/issues/726#issuecomment-561956636). Here is more background

For technical background, see this [StackOverflow question](https://stackoverflow.com/questions/21683607/running-a-single-instance-of-software-for-multiple-selected-files-by-right-click). If there were a better technical solution for the integration with the Windows shell, I would be open to fixing it.

**Comment by armbck** [comment]
Thanks @az0 for your answer. Would [Factor Mystic's solution at Stackoverflow](https://stackoverflow.com/questions/1826791/how-to-pass-in-multiple-file-folder-paths-via-a-right-click-event-verb-to-an-e) be appropriate?

**Comment by az0** [comment]
@armbck 
Yes, it would work, but it feels clunky. That answer was written in 2009, so I am hoping with Windows 10 there is a better way.

**Comment by rados** [comment]
@az0 @abitrolly 
I've also found that context menu command in Windows Explorer doesn't work even for a single file in ver 4.2. It works in ver 4.0 though so I guess it is a kind of regression. I can trace it further.

**Comment by armbck** [comment]
You are correct @abitrolly.
In 4.2.0 shredding with context menu does not work at all
In 4.0.0 it shreds one file even if multiple were selected
In 3.2.0 it amazingly works for multiple files. 

@az0 would there be some code in 3.2.0 that could be used in the upcoming release to fix the issue?

**Comment by rados** [comment]
@az0 
I can trace it on my Windows machine.

**Comment by rados** [comment]
@az0
Is finding a way to shred multiple files / folders a priority?

**Comment by armbck** [comment]
At least to me shredding it is the most used feature of BleachBit.

Btw, if using SendTo-functionality in windows mouse-right-click-menu is easier to implement, that suits fine.

**Comment by az0** [comment]
> @az0
> Is finding a way to shred multiple files / folders a priority?

This is currently not a top priority because there have not been man complaints, and there are several decent workarounds:

1. User can drag and drop. However, Windows does not allow this to happen between and elevated and non-elevated process.
1. User can copy files to clipboard and paste into BleachBit.
1. User can open BleachBit, choose open to shred files/folders, and then go "hunting" for them
1. In some cases, user can set up a custom location through the preferences.

Today I went through the 4.4 milestone and tagged more issues with a high priority.

**Comment by armbck** [comment]
I understand your point of view. The pain grows when having to click several times and needing to focus separately opening BB first (adding to thousands of clicks per day for other reasons), when trying just to quickly delete files securely. Efficiency has a high value. I hope you reconsider this as an investment to user experience!

**Comment by az0** [comment]
@armbck 

I just added a shortcut to make this easier. You can find it in [CI build](https://ci.bleachbit.org/) 1842 or later.

Here are the instructions. Once you can get used to it, it goes quickly.

1. In Windows Explorer, copy the files you want to shred to the clipboard. You can select multiple files using the SHIFT and arrow keys, and the shortcut CTRL+C is the faster than right clicking.
2. Switch to the BleachBit Window. Tip: ALT+TAB is a quick way to switch windows
3. Press CTRL+V on the main BleachBit window.
4. When you see the confirmation dialog, type TAB to switch to the delete button. 
5. Type the ENTER key to choose the delete button. 

Let me know if a tutorial video would help.
